text/0249-v2-experiments.md

```diff
@@ -199,6 +199,82 @@ Disadvantages:
   mitigate this downside by keeping the old unstable package around, but that
   leads to duplicated classes.
 
+### Option 5: superset/subset releases
+
+Instead of *splitting* the stable and experimental APIs into two different
+packages, we release two builds of the same source code, where the *experimental*
+build is a superset of the *stable* build.
+
+This can be done automatically, by using jsii to strip out experimental APIs
+from the **public interface** of the stable release. The implementation
+(`.js` files) will be the same in both releases, so stable APIs in
+`aws-cdk-lib` can safely use experimental APIs internally (though not in
+their public API surfaces).
+
+**How does stripping APIs from the public work?**
+
+| TypeScript            | JavaScript | Java, Python, C#, Go            |
+|-----------------------|------------|---------------------------------|
+| Stripped from `.d.ts` | N/A        | No bindings generated by `jsii` |
+
+In TypeScript, the compiler will prevent usage of APIs that are not advertised
+in the `.d.ts` files. Users will be able to cast this away using `as any` though.
+They will also be able to confuse the compiler enough that they will get to pass
+struct properties that are technically `@experimental`.
+
+For JavaScript, an IDE will (hopefully) not autocomplete APIs missing from
+the `.d.ts` files, but otherwise nothing prevents users from calling the
+APIs, or passing struct properties that are technically experimental. On the
+other hand, JavaScript devs are by definition required to read the docs,
+so they can't miss the *experimental* banners displayed there.
+
+For jsii client languages, the bindings for experimental APIs will simply not
+exist so there's no way to work around that (though we'll have to check
+whether for example Python makes it possible to pass struct values that
+aren't in the declared API surface).
+
+**Vending**
+
+There are a number of different ways we can choose to distinguish the two builds.
+See Appendix D for an evaluation of all the possibilities.
+
+- Distinguish by package name
+  - Can't be used in **pip**: there's no way to have an app that uses the experimental
+    library work with a library that uses the stable library (no way to prevent the
+    transitive dependency package with a different name from being installed).
+- Distinguish by prerelease tag
+  - Can't be used in **Maven**: an app can't depend on stable versions only
+    (a range of `[1.60.0,2.0.0)` would also match `1.61.0-experimental`).
+- Distinguish by major version
+  - All package managers have a way to bring in a different version of a package,
+    although many of them will complain that declared dependencies aren't satisified.
+
+Distinguishing by major version is the only possibility that works across all
+package managers.
```

Comment:
If the conclusion here is "we have to do it by major version", then I would state that here, and move the explanation of why that is into the appendix itself.

Reply:
I think a short explainer with a pointer to the appendix for more details reads better. It's not like it takes up giant swathes of space.